python/ccao/vars_funs.py

```diff
@@ -0,0 +1,155 @@
+    Rename variables from one naming convention to another.
+
+    This function renames columns in a dataset based on a dictionary that maps
+    names from one convention to another. It can rename columns in-place or return
+    a character vector of renamed columns, behavior that is configurable using
+    the `output_type` argument.
+
+    :param data:
+        DataFrame or list of column names to rename.
+        If a DataFrame, renames columns directly.
+    :type data: pandas.DataFrame or list[str]
+
+    :param names_from:
+        The source naming convention to rename from.
+        Must match a key in the dictionary.
+    :type names_from: str
+
+    :param names_to:
+        The target naming convention to rename to.
+        Must match a key in the dictionary.
+    :type names_to: str
+
+    :param output_type:
+        Output type. Either `"inplace"`, which mutates the input data frame,
+        or `"vector"`, which returns a list of strings with the construction
+        new_col_name = old_col_name.
+    :type output_type: OutputType or str
+
+    :param dictionary:
+        The dictionary for mapping column names.
+        Must contain keys like `var_name_<names_from>` and `var_name_<names_to>`.
+    :type dictionary: pandas.DataFrame
+
+    :raises ValueError: If required arguments are invalid or the dictionary does not meet format requirements.
+    :raises TypeError: If `data` is neither a DataFrame nor a list of column names.
+
+    :return:
+        Either the input data with renamed columns if `output_type` is
+        `"inplace"` and the input data is a DataFrame, otherwise a list of
+        renamed columns.
+    :rtype: pandas.DataFrame or list[str]
+
+    :example:
+
+    .. code-block:: python
+
+        import ccao
+
+        ccao.vars_rename(
+            data=["char_yrblt"],
+            names_from="athena",
+            names_to="pretty",
+            output_type="vector"
+        )
+    """
+    # Validate the dictionary schema
+    dictionary = dictionary if dictionary is not None else vars_dict
+    if not isinstance(dictionary, pd.DataFrame) or len(dictionary) == 0:
+        raise ValueError("dictionary must be a non-empty pandas DataFrame")
+
+    # Make sure the dictionary contains variable columns
+    dictionary_var_columns = [
+        col
+        for col in list(dictionary.columns.values)
+        if col.startswith(var_name_prefix)
+    ]
+    if not len(dictionary_var_columns) >= 2:
+        raise ValueError(
+            f"dictionary must contain at least two columns starting with "
+            f"{var_name_prefix}"
+        )
+
+    # Get a list of possible names_from and names_to from dictionary
+    possible_names_args = [
+        col.replace(f"{var_name_prefix}_", "")
+        for col in dictionary_var_columns
+    ]
+
+    # Validate names arguments
+    if not isinstance(names_from, str) or not isinstance(names_to, str):
+        raise ValueError("names_from and names_to must be strings")
```

Comment:
**question: (non-blocking):** Since the function is typed, isn't this check redundant? Same goes for the other `isinstance` checks.